src/App.tsx

```diff
@@ -15,48 +22,123 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const goodsWithKeys: Good[] = goodsFromServer.map((elem) => {
+  return {
+    good: elem,
+    id: uuidv4(),
+  };
+});
+
+enum SortType {
+  Length = 'Length',
+  Alphabetically = 'Alphabetically',
+}
+
+interface SortParams {
+  sortParam: SortType | '';
+  isReversed: boolean;
+}
+
+const getPreparedGoods = (
+  goods: Good[],
+  { sortParam, isReversed }: SortParams,
+): Good[] => {
+  const preparedGoods = [...goods];
+
+  if (sortParam) {
+    preparedGoods.sort(({ good: firstGood }, { good: secondGood }) => {
+      switch (sortParam) {
+        case SortType.Alphabetically:
+          return firstGood.localeCompare(secondGood);
+
+        case SortType.Length:
+          return firstGood.length - secondGood.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [sortParam, setSortParam] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+  const preparedGoods = getPreparedGoods(
+    goodsWithKeys,
+    { sortParam, isReversed },
+  );
+
+  const handleResetSorting = () => {
+    setIsReversed(false);
+    setSortParam('');
+  };
+
+  const isSorted = (sortParam || isReversed);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classes(
+            'button',
+            'is-info',
+            { 'is-light': sortParam !== SortType.Alphabetically },
+          )}
```

Comment:
```suggestion
          className={classes(
            'button is-info',
            { 'is-light': sortParam !== SortType.Alphabetically },
          )}
```

can be more simplify, check all places

Reply:
Thanks